pdns/channel.cc

```diff
@@ -0,0 +1,115 @@
+/*
+ * This file is part of PowerDNS or dnsdist.
+ * Copyright -- PowerDNS.COM B.V. and its contributors
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of version 2 of the GNU General Public License as
+ * published by the Free Software Foundation.
+ *
+ * In addition, for the avoidance of any doubt, permission is granted to
+ * link this program with OpenSSL and to (re)distribute the binaries
+ * produced as the result of such linking.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "channel.hh"
+
+namespace pdns::channel
```

Comment:
That's weird because the CI's `clang-tidy` actually complained about the initial `namespace pdns { namespace channel`, leading me to switch to the new form in https://github.com/PowerDNS/pdns/pull/11427/commits/35b27ac8d18b43702a63e4078b68776c66f561ab : https://clang.llvm.org/extra/clang-tidy/checks/modernize/concat-nested-namespaces.html

Reply:
Ah. I think I figured out why, creating a git worktree with magit - which I usually do to review PRs - doesn't copy over ignored files. So my clang-tidy was complaining using the default set of configuration, in this particular case saying that nested namespace declarations are a C++17 extension.